### PR TITLE
Add redeeming PSBT transaction

### PR DESCRIPTION
**Pull request by tvpeter**
# ↪️ Pull Request

## 💻 What does this PR do?

- [x] Refactor Funding transaction methods into Txn trait
- [x] Add documentation for testing
- [x] Add creating PSBT redeeming transaction

## ✔️ Check List
- [x] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [x] Docs have been added / updated if required
- [ ] Filled out test instructions (In case there aren't any unit tests)

## 🔒 Related Issues

- Closes #

### CODE REVIEW

**Comment by Jeezman** [comment]
@tvpeter looking through. Pls fix lint issues

**Comment by tvpeter** [comment]
> @tvpeter looking through. Pls fix lint issues

I have fixed the linting issues

**Review by Jeezman — COMMENTED** [review]
@tvpeter Good work. Dropped a couple of comments

File: src/utils/transaction_utils.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+};
+use round::round_down;
+
+use super::{
+	bitcoind_rpc::get_outpoint_value, get_feerate::get_mempool_feerate,
+	validate_address::validate_address,
+};
+use crate::{constants::set_network, domain::funding_transaction::PRECISION};
+
+pub fn get_outpoints_total(inputs: &Vec<OutPoint>) -> Result<f64, String> {
```
[the remark this answers is not in the document]
Comment (tvpeter) [reply src/utils/transaction_utils.rs:13]: yes, it is intentional

File: src/utils/transaction_utils.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+		let receiving_address = validate_address(receiving_address, network)?;
+		let change_address = validate_address(change_address, network)?;
+
+		let receiving_script_pubkey_hash = receiving_address.script_pubkey();
+		let change_script_pubkey_hash = change_address.script_pubkey();
+
+		Ok((receiving_script_pubkey_hash, change_script_pubkey_hash))
+	}
+
+	fn hex_amounts(amount: f64, fees: f64, input_total: f64) -> Result<(Amount, Amount), String> {
```
[the remark this answers is not in the document]
Comment (tvpeter) [reply src/utils/transaction_utils.rs]: done

**Pull request merged**